WordPress/Sniffs/WP/DeprecatedParameterValuesSniff.php

```diff
@@ -0,0 +1,231 @@
+					'alt'     => 'the action "clean_site_cache"',
+					'version' => '4.9.0',
+				),
+			),
+		),
+		'do_action_deprecated' => array(
+			1 => array(
+				'blog_details' => array(
+					'alt'     => 'the filter "site_details"',
+					'version' => '4.7.0',
+				),
+				'rest_enabled' => array(
+					'alt'     => 'the rest_authentication_errors filter to restrict access to the REST API',
+					'version' => '4.7.0',
+				),
+			),
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$param_count = count( $parameters );
+		foreach ( $this->target_functions[ $matched_content ] as $position => $paramerter_args ) {
+
+			if ( $position > $param_count ) {
+				break;
+			}
+
+			if ( ! isset( $parameters[ $position ] ) ) {
+				continue;
+			}
+
+			$matched_parameter    = $this->strip_quotes( $parameters[ $position ]['raw'] );
+			$is_dynamic_parameter = $this->phpcsFile->findNext(
+				array( T_CONSTANT_ENCAPSED_STRING, T_ARRAY, T_FALSE, T_TRUE, T_NULL, T_LNUMBER, T_WHITESPACE ),
+				$parameters[ $position ]['start'],
+				$parameters[ $position ]['end'] + 1,
+				true
+			);
+
+			if ( ! $is_dynamic_parameter && ! isset( $paramerter_args[ $matched_parameter ] ) ) {
+				continue;
+			}
+
+			$message        = '%s( %s ) was found.';
+			$stack_position = $stackPtr;
+			$data           = array(
+				$matched_content,
+				$parameters[ $position ]['raw'],
+			);
+
+			if ( $is_dynamic_parameter ) {
+				$message       .= ' The "%s" parameter is dynamic. Please check the value manually.';
+				$stack_position = $is_dynamic_parameter;
+				$data[]         = $matched_parameter;
+			} else {
+				$message       .= ' The "%s" parameter has been deprecated since WordPress version %s.';
+				$stack_position = $this->phpcsFile->findNext(
+					array( T_CONSTANT_ENCAPSED_STRING, T_ARRAY, T_FALSE, T_TRUE, T_NULL, T_LNUMBER ),
+					$parameters[ $position ]['start'],
+					$parameters[ $position ]['end'] + 1
+				);
+				$data[]         = $matched_parameter;
+				$data[]         = $paramerter_args[ $matched_parameter ]['version'];
+			}
+
+			if ( ! empty( $paramerter_args[ $matched_parameter ]['alt'] ) ) {
+				$message .= ' Use %s instead.';
+				$data[]   = $paramerter_args[ $matched_parameter ]['alt'];
+			}
+
+			if ( $is_dynamic_parameter || version_compare( $paramerter_args[ $matched_parameter ]['version'], $this->minimum_supported_version, '>=' ) ) {
+				$this->phpcsFile->addWarning( $message, $stack_position, $matched_content . 'Found', $data );
```

Comment:
Couple of remarks about the error code:
1. If the message about dynamic parameters would stay, it should have a different error code.
2. The `$matched_content` for the error code should be wrapped in a call `$this->string_to_errorcode()`, though having it as is doesn't actually make the error code properly modular as, based on the above array, most of the time, it would still cover two or three different deprecated values.
    I'm not saying the code _should_ be modular, but if you want it to be, then it should be properly modular.